src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -441,6 +447,11 @@ public void Enqueue(IThreadPoolWorkItem callback, bool forceGlobal)
             if (null != tl)
             {
                 tl.workStealingQueue.LocalPush(callback);
+
+                if (Volatile.Read(ref WorkStealingQueueList.wsqActive) != WorkStealingQueueList.WsqNowActive)
```

Comment:
For arm64, the volatile write & read will translate into single store-release and load-acquire instruction.  They should be ordered properly.

For arm32, yoare correct, this a Full Barrier, Store, Load Full Barrier.  So it is broken for arm32.

Maybe better is 

```C#
tl.workStealingQueue.LocalPush(callback);

MemoryBarrier();

if (WorkStealingQueueList.wsqActive != WorkStealingQueueList.WsqNowActive)
{
    Volatile.Write(ref WorkStealingQueueList.wsqActive, WorkStealingQueueList.WsqNowActive);
}
```